.github/scripts/compile.sh

```diff
@@ -133,7 +133,10 @@ do_build () {
 
 # Do a regular build without asan or tsan.
 do_build "" OFF
+# and install Proton Python
+python3 -m pip install --prefix="$PROTON_INSTALL_DIR/usr" "$(find "$PROTON_BUILD_DIR/python/dist/" -name 'python-qpid-proton*.tar.gz')"
```

Comment:
the script starts with

```
# https://sipb.mit.edu/doc/safe-shell
set -Eefuxo pipefail
```

so * expansion does not work... I stumbled upon this myself ;(

Reply:
I think you'd be better using the wheel if you want to avoid needing the compile chain in the run container - the pattern would give you `python_qpid_proton-0.39.0.dev0-cp311-cp311-linux_x86_64.whl` for 0.39-dev0 version for cpython 3.11.
The .tar.gz is more generally useful if you might have a different python in the build container and run container though.

Reply:
Also you might want ot use a multiple roots for find that will work for 0.38 and 0.39 to ease the transition. For 0.38 the path would be `python/pkgs` for 0.39 as you use here its `python/dist`.

Reply:
Sorry I misunderstood this change!
I think it would really be better to copy the pkg to the run container and use the active python there to install - that will make sure that you don't need to worry about where the prefix is. Just make sure that te python that runs `python -m pip install /..../python-qpid-proton...whl` or `....tar.gz` is the same one that will run the scripts that use proton.

Reply:
> I think you'd be better using the wheel if you want to avoid needing the compile chain in the run container

I am quite happy with doing `--prefix="$PROTON_INSTALL_DIR/usr"` and copying that to the run container under /usr (that's what tar is used later for).

> Also you might want ot use a multiple roots for find that will work for 0.38 and 0.39 to ease the transition.

@ganeshmurthy I guess we want that...

> I think it would really be better to copy the pkg to the run container and use the active python there to install

It would be more foolproof, the pip install for the wheel would check that versions of Python match, which here is not checked... On the other hand, the python version is in the site-packages path, so actually, there is sort of check for that. Using wheel means one extra dependency for the build container.

> The .tar.gz is more generally useful if you might have a different python in the build container and run container though.

The wheel could also be multiversion, one day. It is a feature that I think is possible to implement, https://issues.apache.org/jira/browse/PROTON-2678

Reply:
Ah multiversion wheel for linux seems like a very good idea right now - it doesn't seem too hard and that is what you get by default on windows already I'm pretty sure.

Reply:
The wheel python dependency is a very small extra dependency compared to having the compilers and cffi etc.

Reply:
> Ah multiversion wheel for linux seems like a very good idea right now - it doesn't seem too hard and that is what you get by default on windows already I'm pretty sure.

Linux has also the "manylinux" aspect, IIRC. Not sure if we can have that on Linux, though, because that should mean no dependencies on external .so files, and Proton depends on cyrus sasl, ssl, and so on... But anyways, I might be wrong, and it's offtopic here.

Reply:
> The wheel python dependency is a very small extra dependency compared to having the compilers and cffi etc.

I need cffi anyways, I think, isn't that a runtime dep now? And currently in this PR I build the proton package in build container, install it in that --prefix, and tar that and extract in run container (together with the rest of installed proton), so I don't have compiler in run image.

Reply:
I think the abi3 supercedes manylinux. manylinux is tricky for us because it relies on a fixed (and somewhat old) build environment to make sure that the c libraries link to the system glibc.